param/parameterized.py

```diff
@@ -3538,6 +3642,10 @@ def _state_pop(self):
             elif hasattr(g,'state_pop') and isinstance(g,Parameterized):
                 g.state_pop()
 
+    @bothmethod
```

Comment:
Is it supposed to display classes with the rich output? If so that's not what I see:
<img width="123" alt="image" src="https://github.com/holoviz/param/assets/35924738/9a2777a9-c79c-45a0-84c0-5e72900e9b03">

Reply:
Yes, I would expect and hope for rich output for both classes and instances. Maybe a missing `@bothmethod` elsewhere?

Reply:
The current implementation of `bothmethod` was turning the method into a `functools.partial` object whose type is not `MethodType` (not `FunctionType` either). IPython checks the type of the repr function (e.g. `_repr_html_`) on the object to be displayed:

https://github.com/ipython/ipython/blob/9da6376aa74dad72f5a84cdd0bf5ad0f1d809308/IPython/utils/dir2.py#L78-L79

This was failing with the current implementation, which I have replaced in https://github.com/holoviz/param/pull/425/commits/618b88e7f8fe488fbf1994ebfea29af5c8bd80f1 to satisfy IPython's inspection approach.

Reply:
<img width="535" alt="image" src="https://github.com/holoviz/param/assets/35924738/e6f7e1c7-e9a9-48d3-85c5-e6f240e49e2e">